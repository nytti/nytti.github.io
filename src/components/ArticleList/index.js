// Pure component that renders news items in Material UI responsive GRID
// Looks nice on any display size from smartphone to TV

import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

// Material UI
import Grid from '@material-ui/core/Grid'

// Components
import ArticleWrapper from 'components/ArticleWrapper'
import ArticleListEmpty from 'components/ArticleListEmpty'
import ArticlePreview from 'components/ArticlePreview'

export default function ArticleListWrapper({ articles, topic }) {
  if (!articles.length) {
    return <ArticleListEmpty variant="headline" />
  }
  return (
    <Fragment>
      <Grid container spacing={16}>
        {
          articles.map(id => (
            <Grid item xs={12} sm={12} md={6} lg={4} xl={4} key={id}>
              <ArticleWrapper id={id} topic={topic} component={ArticlePreview} />
            </Grid>))
        }
      </Grid>
    </Fragment>
  )
}

ArticleListWrapper.propTypes = {
  articles: PropTypes.array.isRequired,
  topic: PropTypes.string.isRequired,
}
