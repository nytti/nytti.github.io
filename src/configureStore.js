import { createStore, combineReducers, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { routerReducer, routerMiddleware } from 'react-router-redux'
import { gettersMiddleware } from 'utils/reduxGetters'

import * as reducers from './reducers'

export default (history, ...middlewares) => {
  const createReducer = () => combineReducers({
    router: routerReducer,
    ...reducers,
  })

  const store = createStore(
    createReducer(),
    applyMiddleware(
      thunk,
      routerMiddleware(history),
      gettersMiddleware,
      ...middlewares,
    )
  )

  if (module.hot) {
    module.hot.accept('./reducers', () => store.replaceReducer(createReducer()))
  }

  return store
}
