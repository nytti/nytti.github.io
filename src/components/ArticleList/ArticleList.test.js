import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import items from 'articles/items'
import state from 'articles/state'

import ArticleList from './index'

// Mocking store with one article

// Mocking store with test data
const mockStore = configureStore([thunk])
const store = mockStore(state)

describe('<ArticleList />', () => {
  test('<ArticleList /> snapshot', () => {
    const component = shallow(
      <ArticleList
        store={store}
        topic="fashion"
        articles={[items[0]._id]}
      />
    )
    expect(component).toMatchSnapshot()
  })

  test('<ArticleList /> empty snapshot', () => {
    const component = shallow(
      <ArticleList
        store={store}
        topic="fashion"
        articles={[]}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
