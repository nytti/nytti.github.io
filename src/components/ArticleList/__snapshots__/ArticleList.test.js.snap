// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleList /> <ArticleList /> empty snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleListWrapper
    articles={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    topic="fashion"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "variant": "headline",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "variant": "headline",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ArticleList /> <ArticleList /> snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleListWrapper
    articles={
      Array [
        "Item-Id-Test",
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    topic="fashion"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <WithStyles(Grid)
        container={true}
        spacing={16}
      >
        <WithStyles(Grid)
          item={true}
          lg={4}
          md={6}
          sm={12}
          xl={4}
          xs={12}
        >
          <Connect(ArticleWrapper)
            component={[Function]}
            id="Item-Id-Test"
            topic="fashion"
          />
        </WithStyles(Grid)>
      </WithStyles(Grid)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Grid)
            item={true}
            lg={4}
            md={6}
            sm={12}
            xl={4}
            xs={12}
          >
            <Connect(ArticleWrapper)
              component={[Function]}
              id="Item-Id-Test"
              topic="fashion"
            />
          </WithStyles(Grid)>,
        ],
        "container": true,
        "spacing": 16,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Item-Id-Test",
          "nodeType": "class",
          "props": Object {
            "children": <Connect(ArticleWrapper)
              component={[Function]}
              id="Item-Id-Test"
              topic="fashion"
            />,
            "item": true,
            "lg": 4,
            "md": 6,
            "sm": 12,
            "xl": 4,
            "xs": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "id": "Item-Id-Test",
              "topic": "fashion",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <WithStyles(Grid)
          container={true}
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
            lg={4}
            md={6}
            sm={12}
            xl={4}
            xs={12}
          >
            <Connect(ArticleWrapper)
              component={[Function]}
              id="Item-Id-Test"
              topic="fashion"
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Grid)
              item={true}
              lg={4}
              md={6}
              sm={12}
              xl={4}
              xs={12}
            >
              <Connect(ArticleWrapper)
                component={[Function]}
                id="Item-Id-Test"
                topic="fashion"
              />
            </WithStyles(Grid)>,
          ],
          "container": true,
          "spacing": 16,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Item-Id-Test",
            "nodeType": "class",
            "props": Object {
              "children": <Connect(ArticleWrapper)
                component={[Function]}
                id="Item-Id-Test"
                topic="fashion"
              />,
              "item": true,
              "lg": 4,
              "md": 6,
              "sm": 12,
              "xl": 4,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "id": "Item-Id-Test",
                "topic": "fashion",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
